### PR TITLE
*: fix coverage test failures

### PR DESCRIPTION
**Pull request by gyuho**
Fix https://github.com/coreos/etcd/issues/8924.

### CODE REVIEW

**Comment by gyuho** [comment]
@xiang90 

The coverage test is fixed now (https://jenkins-etcd.prod.coreos.systems/job/etcd-coverage/3055/console -- no more panic).

Copied the code to `./tools`, so we don't need to wait for the upstream fix (filed an PR here https://github.com/wadey/gocovmerge/pull/5). Base test container image is already running the fixed `gocovmerge`.

**Comment by xiang90** [comment]
it seems that travis is still failing?

**Comment by gyuho** [comment]
@xiang90 Should pass once https://github.com/coreos/etcd/pull/8971 is merged.

**Comment by gyuho** [comment]
@xiang90 PTAL.

**Comment by xiang90** [comment]
we should avoid putting this into our repo if possible.

**Comment by gyuho** [comment]
@xiang90 I will use my fork then.

**Comment by gyuho** [comment]
@xiang90 PTAL. Some tests are failing due to docker issues, but coverage test panic is resolved.

File: test [hunk 1]
```diff
@@ -202,6 +202,7 @@ function cov_pass {
 	# expects to start with a non-empty file
 	cp "$COVERDIR"/etcdserver.coverprofile "$COVERDIR"/cover.out
 	for f in "$COVERDIR"/*.coverprofile; do
+		echo "Merging ${f}"
```
Comment (xiang90) [inline test]: merging test coverage files.

**Comment by xiang90** [comment]
lgtm if test coverage passes.

**Comment by gyuho** [comment]
Passed https://jenkins-etcd.prod.coreos.systems/job/etcd-coverage/3090/console.

**Pull request merged**